src/entity/user/User.ts

```diff
@@ -183,6 +183,9 @@ type AddNewUserResult =
   | { status: 'failed'; reason: UserFailErrorKeys; error?: Error };
 
 const checkRequiredFields = (data: AddUserData): boolean => {
+  if (data && data.username && !data.experienceLevel) {
```

Comment:
`data?.username` use optional chaining on both?

Reply:
changed to ante's suggestion, should be good as data?.username would never be true if there was no data